src/components/AgronodInputs/AgronodTextField/AgronodTextField.tsx

```diff
@@ -0,0 +1,205 @@
+>;
+
+export interface AgronodTextFieldProps extends AgronodTextFieldBaseProps {
+  status?: "warning" | "error" | "default";
+  emptyStyle?: "highlighted" | "default";
+  helperText?: string;
+  hideHelperText?: boolean;
+  tooltipText?: string;
+  hasIcon?: boolean;
+  label?: string;
+  textAligment?: string;
+  endAndorment?: string | JSX.Element;
+}
+
+const StyledMuiTextField = ({
+  status,
+  emptyStyle,
+  hideHelperText,
+  helperText,
+  hasIcon,
+  textAligment,
+  endAndorment,
+  ...rest
+}: AgronodTextFieldProps) => {
+  const icon = useMemo(() => {
+    if (status === "error" && hasIcon) {
+      return <AgronodIcon name="errorContained" color="error" />;
+    }
+    if (status === "warning" && hasIcon) {
+      return <AgronodIcon name="warningContained" color="warning" />;
+    }
+    return null;
+  }, [status]);
+
+  return (
+    <MuiTextField
+      InputLabelProps={{
+        color: "secondary",
+        shrink: false,
+      }}
+      InputProps={{
+        startAdornment: icon !== null && (
+          <InputAdornment position="start">{icon}</InputAdornment>
+        ),
+        endAdornment: endAndorment !== undefined && (
+          <InputAdornment position="start">{endAndorment}</InputAdornment>
+        ),
+      }}
+      variant="outlined"
+      sx={(theme) => ({
+        "& .MuiOutlinedInput-root:not(.Mui-disabled)": {
+          backgroundColor:
+            status === "warning"
+              ? theme.palette.warning.pastel
+              : status === "error"
+                ? theme.palette.error.pastel
+                : undefined,
+
+          "& fieldset": {
+            borderColor:
+              status === "warning" ? theme.palette.warning.main : undefined,
+          },
+
+          "&:hover fieldset": {
+            borderColor:
+              status === "warning" ? theme.palette.warning.main : undefined,
+          },
+
+          "&.Mui-focused fieldset": {
+            borderColor:
+              status === "warning" ? theme.palette.warning.main : undefined,
+          },
+        },
+        "& .MuiOutlinedInput-input": {
+          textAlign: textAligment ? textAligment : "left",
+        },
+        "& .MuiOutlinedInput-root:has(input[value='']):not(:has(.MuiChip-root)):not(.Mui-disabled)":
+          {
+            backgroundColor:
+              emptyStyle === "highlighted"
+                ? theme.palette.secondary.pastel
+                : undefined,
+          },
+      })}
+      error={status === "error"}
+      helperText={
+        !hideHelperText && helperText !== undefined ? helperText : undefined
+      }
+      {...rest}
+    />
+  );
+};
+
+const AgronodTextField = ({
+  status = "default",
+  emptyStyle = "default",
+  helperText,
+  hideHelperText,
+  tooltipText,
+  hasIcon = false,
+  label,
+  textAligment = "left",
+  endAndorment,
+  ...rest
+}: AgronodTextFieldProps) => {
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  const generateTooltipText = useMemo(() => {
```

Comment:
Cant we always return tooltipText? Why using this condition check?

Reply:
Nice find! it is not necessary, I will fix this